Microsoft.Toolkit.Uwp/Helpers/BluetoothLEHelper/BluetoothLEHelper.cs

```diff
@@ -0,0 +1,350 @@
+        /// <summary>
+        /// Reader/Writer lock for when we are updating the collection.
+        /// </summary>
+        private readonly ReaderWriterLockSlim _readerWriterLockSlim = new ReaderWriterLockSlim();
+
+        /// <summary>
+        /// Advertisement watcher used to find bluetooth devices.
+        /// </summary>
+        private BluetoothLEAdvertisementWatcher _advertisementWatcher;
+
+        /// <summary>
+        /// Device watcher used to find bluetooth devices
+        /// </summary>
+        private DeviceWatcher _deviceWatcher;
+
+        /// <summary>
+        /// The Bluetooth adapter.
+        /// </summary>
+        private BluetoothAdapter _adapter;
+
+        /// <summary>
+        /// Prevents a default instance of the <see cref="BluetoothLEHelper" /> class from being created.
+        /// </summary>
+        private BluetoothLEHelper()
+        {
+            Init();
+        }
+
+        /// <summary>
+        /// Gets the app context
+        /// </summary>
+        public static BluetoothLEHelper Context { get; private set; } = new BluetoothLEHelper();
+
+        /// <summary>
+        /// Gets the list of available bluetooth devices
+        /// </summary>
+        public ObservableCollection<ObservableBluetoothLEDevice> BluetoothLeDevices { get; } = new ObservableCollection<ObservableBluetoothLEDevice>();
+
+        /// <summary>
+        /// Gets a value indicating whether app is currently enumerating
+        /// </summary>
+        public bool IsEnumerating
+        {
+            get
+            {
+                if (_advertisementWatcher == null)
+                {
+                    _advertisementWatcher = new BluetoothLEAdvertisementWatcher();
+                }
+
+                return _advertisementWatcher.Status == BluetoothLEAdvertisementWatcherStatus.Started;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether peripheral mode is supported by this device
+        /// </summary>
+        public bool IsPeripheralRoleSupported => _adapter.IsPeripheralRoleSupported;
+
+        /// <summary>
+        /// Gets a value indicating whether central role is supported by this device
+        /// </summary>
+        public bool IsCentralRoleSupported => _adapter.IsCentralRoleSupported;
+
+        /// <summary>
+        /// Gets a value indicating whether the Bluetooth LE Helper is supported.
+        /// </summary>
+        public bool IsBluetoothLeSupported => ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 4);
+
+        /// <summary>
+        /// An event for when the enumeration is complete.
+        /// </summary>
+        public event EventHandler<EventArgs> EnumerationCompleted;
+
+        /// <summary>
+        /// Starts enumeration of bluetooth device
+        /// </summary>
+        public void StartEnumeration()
+        {
+            if (_advertisementWatcher?.Status == BluetoothLEAdvertisementWatcherStatus.Started)
```

Comment:
This seems like the method could be called multiple times and subscribe to the same events all over again creating a memory leak. Should check if Status is Stopped, etc, and react appropriately